ndcube/mixins/ndslicing.py

```diff
@@ -42,10 +42,13 @@ def _slice(self, item):
         """
         kwargs = super()._slice(item)
 
-        wcs, missing_axis = self._slice_wcs_missing_axis(item)
+        wcs, missing_axis, dropped_coords = self._slice_wcs_missing_axis(item)
         kwargs['wcs'] = wcs
         kwargs['missing_axis'] = missing_axis
-        kwargs['extra_coords'] = self._slice_extra_coords(item, missing_axis)
+        if self._slice_extra_coords(item, missing_axis) is not None:
+            kwargs['extra_coords'] = list(self._slice_extra_coords(item, missing_axis)) + list(dropped_coords)
+        else: 
+            kwargs['extra_coords'] = list(dropped_coords)
```

Comment:
What happens here if ```dropped_coords``` is an empty list and ```self._slice_extra_coords(item, missing_axis)``` is ```None```?  If ```kwargs['extra_coords']``` ends up being an empty list, it should really be set to ```None``` as well.

Also, rather than calling ```self._slice_extra_coords(item, missing_axis)```, twice why not set it to a variable, e.g. 
```python
extra_coords = self._slice_extra_coords(item, missing_axis)
if extra_coords is not None:
    ...
```